src/radical/pilot/utils/component.py

```diff
@@ -758,8 +763,9 @@ def register_output(self, states, output=None):
         sure that the thing is in fact in a final state.
         '''
 
-        if not isinstance(states, list):
-            states = [states]
+        states = ru.as_list(states)
+        if not states:
+            states = None
```

Comment:
should it be `states = [None]`? (similarly as for `register_input`) and also with the next line it is expected `states` to be a `list`

Reply:
Ack, fixed.

Reply:
btw, but in `register_input` method it says `worker handles stateless entities` (and there allowed `if not states: states = [None]`). Is it not applied for `output`? Just want to resolve this and we can push this PR

Reply:
ack, changed.